### DISCUSSION TITLE
PSD to PNG missing alpha channel transparency

### DISCUSSION THREAD

**Discussion by luisfdgon** [ImageMagick/ImageMagick#4417] [Help]
Hi,

I'm trying to convert the following PSD image to PNG:
[porto_wine.psd.zip](https://github.com/ImageMagick/ImageMagick/files/7433022/porto_wine.psd.zip)

The PSD has an alpha channel. I've tried several IM versions 6.9.10 and, 6.9.12 and couldn't get the image converted to PNG with transparency.

Even with a basic 'convert porto_wine.psd porto_wine.png' it generates without transparency. 

In previous version 6.9.7 it was converting fine. Something changed?

Any sugestions? Thank you!

FYI: system is ubuntu 18.04 LTS

**Comment by fmw42** [comment]
Works fine for me on 7.1.0-11 Q16-HDRI Mac OSX Sierra

`magick porto_wine.psd porto_wine.png`

**Reply by luisfdgon** [reply]
Thanks. But what about v6?

**Reply by fmw42** [reply]
It was likely an improvement of the PSD handling in IM 7. I doubt it will be back ported to IM 6, since development there is frozen. But I will leave that to the IM developers to respond officially.

**Reply by dlemstra** [reply]
We will not backport that to IM6.

**Reply by luisfdgon** [reply]
Ok, I understand that. But why did it work on 6.9.7 and stopped working after that and now it works again in IM7?

I was looking into the change log and could not find anything mentioning that. 
I just want to be sure that this was a changed in some 6.9.XX release and that we are not misleading something.

**Reply by dlemstra** [reply]
There have probably been some changes in how we detect the transparency between those version. And we probably made some fixes for other files and broke it for this one. I might take a look at it this weekend and see if this is an easy fix.

**Reply by dlemstra** [reply]
Turns out this was an easy fix that will be resolved in the next version of ImageMagick 6.

**Reply by luisfdgon** [reply]
That's good news. Thank you dlemstra for your effort!
I see in the commits, your fix will be included in 6.9.12-29. When is expected to be released? 

Also, Is there any migration guide from v6 to v7, where I can see breaking changes and plan further migration? 

Thanks again.

**Reply by fmw42** [reply]
See https://imagemagick.org/script/porting.php#cli
